src/io/flutter/preview/PreviewView.java

```diff
@@ -322,6 +332,51 @@ private void selectPath(TreePath selectionPath, boolean focusEditor) {
         currentEditor.getCaretModel().addCaretListener(caretListener);
       }
     }
+
+    if (SHOW_PREVIEW_AREA) {
+      final Element buildMethodElement = getBuildMethodElement(selectionPath);
+      previewAreaPanel.updatePreviewElement(getElementParentFor(buildMethodElement), buildMethodElement);
+    }
+  }
+
+  // TODO: Add parent relationship info to FlutterOutline instead of this O(n^2) traversal.
```

Comment:
Other alternatives.

We could remember the top-level FlutterOutline, then we would need recursive visit, but O(n).

Or we could fill childToParentMap after receiving the top-level outline and rendering it into the tree.

Reply:
We could populate a `parent` field when re-serializing a FlutterOutline?

Reply:
`FlutterOutline` is code generated.
We still could tweak the generator though.

Reply:
Gotcha.

I don't think this will present a performance issue in practice, but would be nice cleanup at some point in the future.